addons/survey/models/survey_survey.py

```diff
@@ -568,6 +568,32 @@ def _get_number_of_attempts_lefts(self, partner, email, invite_token):
     # QUESTIONS MANAGEMENT
     # ------------------------------------------------------------
 
+    def _remove_invalid_conditional_questions(self, questions):
+        possible_trigger_questions = self.env['survey.question'].search([
+            ('is_page', '=', False),
+            ('question_type', 'in', ['simple_choice', 'multiple_choice']),
+            ('suggested_answer_ids', '!=', False),
+            ('survey_id', '=', self.id)
+        ])
+        valid_questions = self.env['survey.question']
+        invalid_questions = self.env['survey.question']
+        for question in questions:
+            if not question.is_conditional:
+                valid_questions |= question
+                continue
+            if question.triggering_question_id in invalid_questions:
+                invalid_questions |= question
+                continue
+            allowed_triggering_question_ids = possible_trigger_questions.filtered(
+                lambda q: q.sequence < question.sequence or q.sequence == question.sequence and q.id < question.id
+            )
+            if (question.triggering_question_id
+                    and question.triggering_question_id.id in allowed_triggering_question_ids.ids):
```

Comment:
`X.id in X.ids` is also not required here, `question.triggering_question_id in allowed_triggering_question_ids` should work fine.